### PR TITLE
Overwrite existing values in MapInstance.Set().

### LINKED ISSUE #63: MapInstance.Set() does not overwrite existing values
Hi,

when running the following code in Firefox, Chrome or other browsers:
```javascript
new Map().set(1, 2).set(0, 9).set(1, 8).get(1);
```
the return value is `8` (as `set(1, 8)` should overwrite the existing entry with value `2`).
However, on Jurassic it returns `2`.

`MapInstance.Set()` looks like the following:
```c#
    [JSInternalFunction(Name = "set")]
    public MapInstance Set(object key, object value)
    {
        if (this.store.ContainsKey(key))
            return this;
        if (key is double && TypeUtilities.IsNegativeZero((double)key))
            key = 0;
        var node = this.list.AddLast(new KeyValuePair<object, object>(key, value));
        this.store.Add(key, node);
        return this;
    }
```

So it checks whether the key already exists, but in that case does not set the new value.
ES2015 says at [23.1.3.9](http://www.ecma-international.org/ecma-262/6.0/#sec-map.prototype.set):
> 5. Repeat for each Record {[[key]], [[value]]} p that is an element of entries,
>    a. If p.[[key]] is not empty and SameValueZero(p.[[key]], key) is true, then
>        i. Set p.[[value]] to value.
>        ii. Return M.

So it seems the first `if` branch misses the code to set the new value.

Thanks!

### PR DESCRIPTION
**Pull request by kpreisser**
Fixes #63

### CODE REVIEW

File: Unit Tests/Library/MapTests.cs [hunk 1]
```diff
@@ -175,14 +175,22 @@ public void keys()
...
             // length
             Assert.AreEqual(2, Evaluate("Map.prototype.set.length"));
+
+            Execute("var i = new Map().set(1, 2).set(0, 9).set(1, 8);");
+
+            // Overwrites values
+            Assert.AreEqual(8, Evaluate("i.get(1)"));
+
+            // Insertion order
+            Assert.AreEqual("1,0", Evaluate("var result = []; i.forEach(function(e1, e2, S) { result.push(e1); }); result.toString();"));
```
Comment (paulbartrum) [inline Unit Tests/Library/MapTests.cs]: This should evaluate to "8,9" (e1 is the value, not the key).

Comment (paulbartrum) [inline Unit Tests/Library/MapTests.cs]: Can you also add an assert for result.push(e2)?  (It should equal "1,0")

**Comment by paulbartrum** [comment]
I just realised that your unit test passed... that means Map.prototype.forEach is broken too.  Darn :-)

**Comment by paulbartrum** [comment]
I've fixed the problem with the forEach parameter ordering and then merged your change.  Thanks!

**Comment by kpreisser** [comment]
Sorry, I didn't notice the issue with `forEach`. Thanks for fixing it!

**Pull request merged**